src/plugins/pane-primitive-wrapper.ts

```diff
@@ -0,0 +1,43 @@
+import { IPaneApi } from '../api/ipane-api';
+import { IPanePrimitive } from '../api/ipane-primitive-api';
+
+import { DeepPartial } from '../helpers/strict-type-checks';
+
+import { IPrimitiveWrapper } from './primitive-wrapper-base';
+
+/**
+ * Interface for a primitive with options.
+ */
+export interface IPanePrimitiveWithOptions<T, K> extends IPanePrimitive<T> {
+	/**
+	 * @param options - Options to apply. The options are deeply merged with the current options.
+	 */
+	applyOptions?: (options: DeepPartial<K>) => void;
+}
+
+export class PanePrimitiveWrapper<T, Options = unknown, TPrimitive extends IPanePrimitiveWithOptions<T, Options> = IPanePrimitive<T>> implements IPrimitiveWrapper<T, Options> {
+	protected _primitive: TPrimitive;
+	protected _pane: IPaneApi<T>;
```

Comment:
Lets try use `private` where possible and only use `protected` if we need to.
```suggestion
	private _primitive: TPrimitive;
	private _pane: IPaneApi<T>;
```